appservice/src/createAppService/AppInsightsLocationStep.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceManagementClient } from "azure-arm-resource";
+import { Provider, ProviderResourceType } from "azure-arm-resource/lib/resource/models";
+import { Location } from "azure-arm-resource/lib/subscription/models";
+import { AzureWizardPromptStep, createAzureClient, IAzureQuickPickItem, IAzureQuickPickOptions } from "vscode-azureextensionui";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { nonNullProp } from "../utils/nonNull";
+import { requestUtils } from "../utils/requestUtils";
+import { IAppServiceWizardContext } from "./IAppServiceWizardContext";
+
+export class AppInsightsLocationStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
+
+    public async prompt(wizardContext: IAppServiceWizardContext): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: localize('locationAppInsights', 'Select a location for the new App Insights component.'), id: `AppInsightsLocationStep/${wizardContext.subscriptionId}` };
+        wizardContext.newAppInsightsLocation = (await ext.ui.showQuickPick(this.getQuickPicks(wizardContext), options)).label;
+    }
+
+    public shouldPrompt(wizardContext: IAppServiceWizardContext): boolean {
+        return !wizardContext.appInsightsComponent && !wizardContext.newAppInsightsLocation;
+    }
+
+    public async getSupportedLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<string | undefined> {
```

Comment:
I don't see `getSupportedLocation` called in this file, so it probably shouldn't be in here

Reply:
`getSupportedLocation` isn't called in this file, but it calls two other functions that are.  Of course, this might all be refactored anyway.